### DISCUSSION TITLE
Embedded key/value store?

### DISCUSSION THREAD

**Discussion by hrstoyanov** [belaban/jgroups-raft#217] [General]
Might be missing something, but is there any particular reason to choose a non-java embedded key/store engine (LevelDB)? There appear to be a few java-based, such as:

https://github.com/yahoo/HaloDB

https://github.com/OpenHFT/Chronicle-Queue

**Comment by belaban** [comment]
At the time I started jgroups-raft, LevelDB was recommended to me, and it's been relatively painless (no major bugs) to use. Meanwhile, we have a simple and fast file store that will be (or is already) the default store: [1].

[1] https://github.com/belaban/jgroups-raft/tree/master/src/org/jgroups/raft/filelog

**Reply by hrstoyanov** [reply]
Thank you!
Maybe you can make LevelDB optional then, then you can mark it as optional in the pom.xml
